metadata/src/main/java/org/apache/kafka/controller/FeatureControlManager.java

```diff
@@ -343,10 +343,11 @@ private ApiError updateMetadataVersion(
             if (!metadataChanged) {
                 log.warn("Downgrading metadata.version from {} to {}.", currentVersion, newVersion);
             } else if (allowUnsafeDowngrade) {
-                return invalidMetadataVersion(newVersionLevel, "Unsafe metadata downgrade is not supported.");
+                return invalidMetadataVersion(newVersionLevel, "Unsafe metadata downgrade is not supported " +
+                        "in this version.");
```

Comment:
Any reason why we added it back?

Reply:
This is something the original KIP calls out and will be eventually implemented. This is also a perfect match with the updated documentation.